components/tidb_query_datatype/src/codec/data_type/bit_vec.rs

```diff
@@ -282,6 +288,41 @@ mod tests {
         assert_eq!(x.len(), 0);
     }
 
+    #[test]
+    fn test_truncate_greater_than_64() {
+        let mut x = BitVec::with_capacity(0);
+        for i in 0..70 {
+            x.push(i % 2 == 0);
+        }
+
+        assert_eq!(x.len(), 70);
+        x.truncate(67);
+        assert_eq!(x.len(), 67);
+        assert_eq!(x.data.len(), 2);
+
```

Comment:
To test if truncate functions correctly, I would recommend push new elements to `BitVec` and check if they are NULL or not.